e2e/create-key.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { test, expect } from "@playwright/test";
+import { randomUUID } from "crypto";
+
+function generateUniqueKeyName() {
+  return `my-test-key-${randomUUID().slice(0, 8)}`;
+}
+
+function generateUniqueKeyDescription() {
+  return `my-test-key-description-${randomUUID().slice(0, 8)}`;
+}
+
+test.describe.serial("API Key Management", () => {
+  // Generate unique key name and description at the describe level
+  const keyName = generateUniqueKeyName();
+  const keyDescription = generateUniqueKeyDescription();
+
+  test.beforeEach(async ({ page }) => {
+    // select tenant
+    await page.goto("/");
+    await page.getByRole("combobox").click();
+    await page.getByRole("option", { name: /default-team.*/ }).click();
+  });
+
+  test("create key", async ({ page }) => {
+    // assert we are on keys page using its url
+    await expect(page).toHaveURL("/keys");
+
+    // create key
+    await page.getByRole("link", { name: "Create New Key" }).click();
+    await page.getByRole("textbox", { name: "Name" }).click();
+    await page.getByRole("textbox", { name: "Name" }).fill(keyName);
+    await page.getByRole("textbox", { name: "Description" }).click();
+    await page
+      .getByRole("textbox", { name: "Description" })
+      .fill(keyDescription);
+    await page.getByRole("combobox", { name: "Expiry" }).click();
+    await page.getByRole("option", { name: "90 days" }).click();
+    await page.getByRole("button", { name: "Create" }).click();
+
+    // assert api key created
+    await expect(page.getByRole("main")).toContainText("API Key Created");
```

Comment:
isn't these text have high chances to be getting changed so shouldn't we use test id instead just checking for text because its e2e not a unit test.